lib/components/narrative/narrative-itineraries.js

```diff
@@ -178,12 +179,14 @@ class NarrativeItineraries extends Component {
     return (
       <ItineraryBody
         active={active}
+        activeItineraryTimeIndex={activeItineraryTimeIndex}
         activeLeg={activeLeg}
         activeStep={activeStep}
         expanded={showDetails}
         index={itinerary.index}
         itinerary={itinerary}
-        key={itinerary.index}
+        // Ensure we update if either index changes
+        key={(activeItineraryTimeIndex + 1) * (itinerary.index + 1)}
```

Comment:
Are these guaranteed to always be unique? Index a=1, b=0 and index a=0,b=1 produce the same key value with this code.

How about a template literal like `${activeItineraryTimeIndex} ${itinerary.index}`
Or you could multiply one by 100 and then add them, since I doubt either index will ever reach 100. 
Or use a binary left shift `<< 8` on one and then binary or them. IDK if that is more efficient in Javascript.

Reply:
I love the binary operator! Thanks for that suggestion works great and avoids these strange edge cases!